rubrics/impl/src/main/java/org/sakaiproject/rubrics/logic/RubricsServiceImpl.java

```diff
@@ -813,6 +813,87 @@ public void deleteRubricAssociationsByItemIdPrefix(String itemId, String toolId)
         }
     }
 
+	public void softDeleteRubricAssociationsByItemIdPrefix(String itemId, String toolId) {
+		try{
+			TypeReferences.ResourcesType<Resource<ToolItemRubricAssociation>> resourceParameterizedTypeReference =
+					new TypeReferences.ResourcesType<Resource<ToolItemRubricAssociation>>() {};
+
+			URI apiBaseUrl = new URI(serverConfigurationService.getServerUrl() + RBCS_SERVICE_URL_PREFIX);
+			Traverson traverson = new Traverson(apiBaseUrl, MediaTypes.HAL_JSON);
+
+			Traverson.TraversalBuilder builder = traverson.follow("rubric-associations", "search", "by-item-id-prefix");
+
+			HttpHeaders headers = new HttpHeaders();
+			headers.add("Authorization", String.format("Bearer %s", generateJsonWebToken(toolId)));
+			builder.withHeaders(headers);
+
+			Map<String, Object> parameters = new HashMap<>();
+			parameters.put("toolId", toolId);
+			parameters.put("itemId", itemId);
+
+			Resources<Resource<ToolItemRubricAssociation>> associationResources = builder.withTemplateParameters(parameters).toObject(resourceParameterizedTypeReference);
+
+			for (Resource<ToolItemRubricAssociation> associationResource : associationResources) {
+                String associationHref = associationResource.getLink(Link.REL_SELF).getHref();
+                deleteRubricEvaluationsForAssociation(associationHref, toolId);
```

Comment:
If this is a soft delete, should we be deleting the evaluations? If we then recover the associated item, won't we want to get the evaluations back, too?